pcdsdevices/areadetector/plugins.py

```diff
@@ -21,6 +21,14 @@ class PluginBase(ophyd.plugins.PluginBase, ADBase):
     """
     enable = C(EpicsSignal, 'EnableCallbacks_RBV.RVAL', write_pv="EnableCallbacks", string=False)
 
+    def __init__(self, *args, **kwargs):
+        # Avoid ophyd PluginBase's init
+        ADBase.__init__(self, *args, **kwargs)
```

Comment:
Dislike this. Divorcing ourselves from the constructor emit seems drastic to avoid one `caget`. I think this is really a problem with the fact that this kind of behavior doesn't belong ing `PluginBase.__init__` and it should be fixed on the `ophyd` end not patched here.

Reply:
It avoids two cagets btw

Reply:
I'm willing to submit the less obvious one as a PR:
`'cam' in self.parent.component_names` vs `hasattr(self.parent, 'cam')`

But it's hard for me to find words to justify not checking the plugin type

Reply:
I mean.  I think the main problem is that the `plugin_type` not matching is an issue in the class construction. Why are we checking this on every `__init__`?

Reply:
it's not the getting that's bad, it's the `wait_for_connection` that gunks up the whole initialization